kv: fix redaction in split logging

nvb opened the pull request:
This commit improves the redaction in `"initiating a split of this range ..."` logs. Previously, these logs would omit the entire "reason" and "extra" fields, which hindered their utility. This log line is important for debugging, so this commit improves it to redact only the necessary information (almost nothing).

The redacted log lines look like:
```
I231109 05:39:54.401617 59582 kv/kvserver/replica_command.go:440 ⋮ [T1,Vsystem,n1,split,s1,r95/1:‹×›] 1135  initiating a split of this range at key /Tenant/2/Table/114/1/‹×› [r96] (load at key /Tenant/2/Table/114/1/‹×› (cpu 906ms, 2268.57 batches/sec, 2261.55 raft mutations/sec))
```
or like:
```
I231109 05:41:29.113805 73695 kv/kvserver/replica_command.go:440 ⋮ [T1,Vsystem,n1,split,s1,r95/1:‹×›] 7390  initiating a split of this range at key /Tenant/2/Table/114/1/‹×› [r101] (512 MiB above threshold size 512 MiB)
```

The second commit then does the same for range merges.

Spotted during a support investigation.

Epic: None
Release note: None

nvb committed:
kv: fix redaction in split logging

This commit improves the redaction in "initiating a split of this range ..."
logs. Previously, these logs would omit the entire "reason" and "extra" fields,
which hindered their utility. This log line is important for debugging, so this
commit improves it to redact only the necessary information (almost nothing).

The redacted log lines look like:
```
I231109 05:39:54.401617 59582 kv/kvserver/replica_command.go:440 ⋮ [T1,Vsystem,n1,split,s1,r95/1:‹×›] 1135  initiating a split of this range at key /Tenant/2/Table/114/1/‹×› [r96] (load at key /Tenant/2/Table/114/1/‹×› (cpu 906ms, 2268.57 batches/sec, 2261.55 raft mutations/sec))
```
or like:
```
I231109 05:41:29.113805 73695 kv/kvserver/replica_command.go:440 ⋮ [T1,Vsystem,n1,split,s1,r95/1:‹×›] 7390  initiating a split of this range at key /Tenant/2/Table/114/1/‹×› [r101] (512 MiB above threshold size 512 MiB)
```

Spotted during a support investigation.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114123)
<!-- Reviewable:end -->

nvb committed:
kv: fix redaction in merge logging

This commit is like the previous commit, but for range merges instead of
range splits. It improves the redaction so that the "reason" field will
be usable when debugging cloud clusters.

Epic: None
Release note: None

miraradeva reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks for the quick fix!

> Reviewed 7 of 7 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114123#-:-NioL2LiCQMcoDjublbj:b-yq9ynn)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kvoli)



<!-- Sent from Reviewable.io -->

kvoli reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114123#-:-NioZMZmCQ68oY-Kz3EA:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

nvanbenschoten commented:
TFTRs!

bors r+
